### PR TITLE
Issue #536 New make target for PR check

### LINKED ISSUE #536: Add `make prcheck`
#### Pre-check

* [x] This is not a IPFS Cluster website content issue ([file those here](https://github.com/ipfs/ipfs-cluster-website/issues))
* [x] I read the [troubleshooting section](https://cluster.ipfs.io/documentation/troubleshooting/) of the website and it did not help
* [x] I searched for similar issues in the repo without luck
* [x] All my peers are running the same cluster version
* [x] All my peers are configured using the same cluster `secret`

#### Basic information

* [x] Version information (mark as appropiate):
  * [x] Master
  * [ ] Release candidate for next version
  * [ ] Latest stable version
  * [ ] An older version I should not be using
* [x] Type (mark as appropiate):
  * [ ] Bug
  * [ ] Feature request
  * [x] Enhancement
* [x] Operating system (mark as appropiate):
  * [x] Linux
  * [ ] macOS
  * [ ] Windows
  * [ ] Other: `which?`
* [x] Installation method (mark as appropiate):
  * [ ] Binaries from dist.ipfs.io
  * [x] Built from sources
  * [ ] Docker
  * [ ] Snap
  * [ ] Other: `which?`

#### Description
Going through a Travis run I see that we do the following
```
make deps
make check
make service
make ctl
make test
```

It would be nice to have a single make target that I can run and check if I am ready to push my PR, say `make prcheck`

### PR DESCRIPTION
**Pull request by kishansagathiya**
This PR adds a single make target that can run and check if I am
ready to push my PR, i.e, `make prcheck`

Fixes #536

### CODE REVIEW

**Comment by kishansagathiya** [comment]
jenkins/pr-merge check fails with test failure of TestClustersPeerRemoveSelf. A re-run of checks should fix this.
I think this is a flaky test since I am seeing this second time.

File: Makefile [hunk 1]
```diff
@@ -108,4 +108,6 @@ docker:
 	docker exec tmp-make-cluster-test sh -c "ipfs-cluster-service -v"
 	docker kill tmp-make-cluster-test
 
+prcheck: deps check service ctl test intall test_sharness
```
Comment (hsanjuan) [inline Makefile]: type: `intall`. But a test target should not modify the running system. Perhaps safest to run without `test_sharness`.

Comment (kishansagathiya) [reply Makefile]: What a horrible typo! :see_no_evil: 
Let me change this a little later today. Will take a look at #539 as well while on it.

Comment (kishansagathiya) [reply Makefile]: > But a test target should not modify the running system.

I think most developers tend to use a separate $GOPATH for development. I personally use a different $GOPATH for every project. And during development anyway developers would be installing the modified version. I think we can keep `install` and `test_sharness`

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 094a4ca14214bb071d23891f4b532d7536f62ab3
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by kishansagathiya** [comment]
Green at last 
Have added fix for #539 as well

File: Makefile (hunk 1 above)
Comment (hsanjuan) [reply Makefile]: Don't assume what people do. I don't do any of those things. And your GOBIN path can be set to anything even if you're on a different GOPATH. Only `make install` should overwrite anything outside the project folder.

Comment (kishansagathiya) [reply Makefile]: Made those changes https://github.com/ipfs/ipfs-cluster/pull/542/commits/7d9c12dd9a34ca467a80f6a7e9513cd954592efa

**Comment by hsanjuan** [comment]
@kishansagathiya btw, since you are a project's collaborator now, you can create and push branches directly to `ipfs/ipfs-cluster` (instead of your fork). This comes handy sometimes, if several of us want to push to the same branch in a PR, and it saves you having to sync your upstream to our upstream, but it's not a must. Just FYI.

**Pull request merged**